src/HexagonOptimize.cpp

```diff
@@ -678,6 +678,25 @@ class OptimizePatterns : public IRMutator {
                 expr = mutate(op->a + neg_b);
                 return;
             } else {
+
+                // No widening subtracts for the expressions of the form:
```

Comment:
I think the best way to handle this might be to add "implementations" of the widening vsub to hvx_64.ll and hvx_128.ll, and add patterns for them. It's a bit more code, but IMHO it's less error prone.

Also - seems like this change should may be in a separate PR? It seems independent of the vtmpy stuff.